src/RestApi/StoreApi/Controllers/CartOrder.php

```diff
@@ -0,0 +1,362 @@
+				$reserved_stock = $wpdb->get_var(
+					$wpdb->prepare(
+						"
+						SELECT SUM( stock_table.`stock_quantity` ) FROM $wpdb->wc_reserved_stock stock_table
+						LEFT JOIN $wpdb->posts posts ON stock_table.`order_id` = posts.ID
+						WHERE stock_table.`product_id` = %d
+						AND posts.post_status = 'wc-draft'
+						AND stock_table.`order_id` != %d
+						AND stock_table.`timestamp` > ( NOW() - INTERVAL 10 MINUTE )
+						",
+						$stocked_product_id,
+						$order->get_id()
+					)
+				);
+
+				// Deals with legacy stock reservation which the core Woo checkout performs.
+				$reserved_stock += ( $hold_stock_minutes > 0 ) ? wc_get_held_stock_quantity( $product ) : 0;
+
+				if ( ( $product->get_stock_quantity() - $reserved_stock - $stock_to_reserve[ $stocked_product_id ] ) < $item->get_quantity() ) {
+					throw new RestException(
+						'woocommerce_rest_cart_order_product_not_enough_stock',
+						sprintf(
+							/* translators: %s: product name */
+							__( 'Not enough units of %s are available in stock to fulfil this order.', 'woo-gutenberg-products-block' ),
+							$product->get_name()
+						),
+						403
+					);
+				}
+
+				// Queue this reservation for later DB insertion.
+				$stock_to_reserve[ $stocked_product_id ] += $item->get_quantity();
+			}
+
+			$stock_to_reserve = array_filter( $stock_to_reserve );
+
+			if ( $stock_to_reserve ) {
+				$stock_to_reserve_rows = [];
+
+				foreach ( $stock_to_reserve as $product_id => $stock_quantity ) {
+					$stock_to_reserve_rows[] = '(' . esc_sql( $order->get_id() ) . ',"' . esc_sql( $product_id ) . '","' . esc_sql( $stock_quantity ) . '")';
+				}
+
+				$values = implode( ',', $stock_to_reserve_rows );
+
+				$wpdb->query(
+					"REPLACE INTO {$wpdb->wc_reserved_stock} ( order_id, product_id, stock_quantity ) VALUES {$values};" // phpcs:ignore
+				);
+			}
+		} catch ( RestException $e ) {
+			return new RestError( $e->getErrorCode(), $e->getMessage(), array( 'status' => $e->getCode() ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Create order and set props based on global settings.
+	 *
+	 * @param RestRequest $request Full details about the request.
+	 * @return \WC_Order A new order object.
+	 */
+	protected function create_order_from_cart( $request ) {
+		$order = $this->get_order_object();
+		$order->set_status( 'draft' );
+		$order->set_created_via( 'store-api' );
+		$order->set_currency( get_woocommerce_currency() );
+		$order->set_prices_include_tax( 'yes' === get_option( 'woocommerce_prices_include_tax' ) );
+		$order->set_customer_id( get_current_user_id() );
+		$order->set_customer_ip_address( \WC_Geolocation::get_ip_address() );
+		$order->set_customer_user_agent( wc_get_user_agent() );
+
+		$this->set_props_from_cart( $order, $request );
+		$this->create_line_items_from_cart( $order, $request );
+		$this->set_props_from_request( $order, $request );
+
+		$order->save();
```

Comment:
Do we need any error handling if the order doesn't save here?

Reply:
Nothing to handle; core logs a warning and returns the ID. There is no way to add extra error handling here.